### PR TITLE
server: Add metrics

### PR DESCRIPTION
**Pull request by shenli**
Add some metrics for server pkg.
@siddontang @coocood @tiancaiamao @zimulala PTAL

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -530,6 +532,17 @@ func (cc *clientConn) handleLoadData(loadDataInfo *executor.LoadDataInfo) error
 
 func (cc *clientConn) handleQuery(sql string) (err error) {
 	startTs := time.Now()
+	defer func() {
+		// Add metrics
+		t := time.Since(startTs)
+		queryHistgram.Observe(float64(t))
+		lable := querySucc
```
Comment (coocood) [inline server/conn.go]: lable -> label

**Comment by ngaut** [comment]
PTAL @overvenus @siddontang

File: server/conn.go [hunk 2]
```diff
@@ -530,6 +532,17 @@ func (cc *clientConn) handleLoadData(loadDataInfo *executor.LoadDataInfo) error
 
 func (cc *clientConn) handleQuery(sql string) (err error) {
 	startTs := time.Now()
+	defer func() {
+		// Add metrics
+		t := time.Since(startTs)
+		queryHistgram.Observe(float64(t))
```
Comment (siddontang) [inline server/conn.go]: use t.Seconds()

Comment (shenli) [inline server/conn.go]: second is a too large unit for most query response time.

**Comment by shenli** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
@siddontang PTAL

File: server/conn.go (hunk 2 above)
Comment (siddontang) [inline server/conn.go]: no, you name the metric with `handle_query_duration_seconds `, so we must use Seconds here.
Btw, `Seconds()` doesn't return only seconds, it returns like 1.1111111 , see https://golang.org/pkg/time/#Duration.Seconds

Comment (shenli) [inline server/conn.go]: But most of out query will be handled in less than 30ms. I will rename it to handle_query_duration_nanoseconds.

Comment (siddontang) [inline server/conn.go]: No no, seconds is enough for ms.

Comment (shenli) [inline server/conn.go]: @siddontang PTAL

File: server/metrics.go [hunk 3]
```diff
@@ -22,8 +22,8 @@ var (
 		prometheus.HistogramOpts{
 			Namespace: "tidb",
 			Subsystem: "server",
-			Name:      "handle_query_duration_seconds",
-			Help:      "Bucketed histogram of processing time (s) of handled queries.",
+			Name:      "handle_query_duration_nanoseconds",
+			Help:      "Bucketed histogram of processing time (ns) of handled queries.",
 			Buckets:   prometheus.ExponentialBuckets(0.0005, 2, 13),
```
Comment (siddontang) [inline server/metrics.go]: I prefer Seconds here, like all other programs (etcd, pd) using prometheus.
And here the buckets is 0.0005, it is for seconds, means the lease is 0.5ms, if you use nano, too small.

**Pull request merged**